src/SarifCli.FunctionalTests/TestData/DoNotUseFriendlyNameAsRuleId/RuleIdMatchesFriendlyName_Expected.sarif

```diff
@@ -0,0 +1,83 @@
+{
+  "$schema": "http://json.schemastore.org/sarif-1.0.0",
+  "version": "1.0.0",
+  "runs": [
+    {
+      "id": "0ecc3107-1021-4ce7-87ff-a553d00bb644",
+      "tool": {
+        "name": "Sarif",
+        "fullName": "Sarif 1.5.28.0",
+        "version": "1.5.28.0",
+        "semanticVersion": "1.5.28",
+        "sarifLoggerVersion": "1.5.28.0",
+        "language": "en-US",
+        "properties": {
+          "Comments": "Object model and utilities for reading and writing files in the SARIF format.",
+          "CompanyName": "Microsoft Corporation",
+          "ProductName": "Microsoft SARIF SDK"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SV0001",
+          "level": "warning",
+          "formattedRuleMessage": {
+            "formatId": "DefaultFormatId",
+            "arguments": [
+              "RuleIdMatchesFriendlyName.sarif",
+              "ImportantRule"
+            ]
+          },
+          "locations": [
+            {
+              "analysisTarget": {
+                "uri": "file:///G:/Code/sarif-sdk/bld/bin/SarifCli.FunctionalTests/AnyCPU_Release/TestData/DoNotUseFriendlyNameAsRuleId/RuleIdMatchesFriendlyName.sarif",
+                "region": {
+                  "startLine": 13,
+                  "startColumn": 27
+                }
+              }
+            }
+          ]
+        }
+      ],
+      "toolNotifications": [
+        {
+          "id": "MSG001.AnalyzingTarget",
+          "physicalLocation": {
+            "uri": "file:///G:/Code/sarif-sdk/bld/bin/SarifCli.FunctionalTests/AnyCPU_Release/TestData/DoNotUseFriendlyNameAsRuleId/RuleIdMatchesFriendlyName.sarif"
+          },
+          "message": "Analyzing 'RuleIdMatchesFriendlyName.sarif'...",
+          "level": "note",
+          "time": "2016-08-23T22:55:51.352Z"
+        }
+      ],
+      "rules": {
+        "SV0001": {
+          "id": "SV0001",
+          "name": "DoNotUseFriendlyNameAsRuleId",
+          "shortDescription": "Do not use the same string for a rule's id and name properties. The id property ...",
+          "fullDescription": "Do not use the same string for a rule's id and name properties. The id property must be a stable, opaque identifer such as \"SV0001\". The name property should be a string that is understandable to an end user, such as \"DoNotUserFriendlyNameAsRuleId\".",
+          "messageFormats": {
+            "DefaultFormatId": "In \"{0}\", the name and id properties of rule \"{1}\" are the same."
```

Comment:
This is IMO an unfortunate feature of the framework: you must leave a placeholder for the target file name, even though it's mentioned everywhere you might look at the result: in the Error List, in the `result` element of the raw log file, and on the console, resulting in this repetition:

> `G:\Code\sarif-sdk\src\SarifCli.FunctionalTests\TestData\DoNotUseFriendlyNameAsRuleId\RuleIdMatchesFriendlyName.sarif(13,27): warning SV0001: In "RuleIdMatchesFriendlyName.sarif", the name and id properties of rule "ImportantRule" are the same.`

This makes sense in BinSkim, where the message all look like "foo.exe is a 64-bit imange with...". But for other tools I don't think it makes sense. This seems to be a bit of tail-wagging-dog, where a convenience feature for BinSkim influenced the framework.

I think BinSkim should just specify {0} manually, and the framework shouldn't provide it.

Or at the very least, if the framework insists on providing the target file name as `{0}`, it shouldn't have an assertion that insists that every parameter be mentioned in the message.